crates/astria-sequencer/src/ibc/ics20_withdrawal.rs

```diff
@@ -106,6 +112,37 @@ async fn establish_withdrawal_target<S: StateRead>(
 impl ActionHandler for action::Ics20Withdrawal {
     async fn check_stateless(&self) -> Result<()> {
         ensure!(self.timeout_time() != 0, "timeout time must be non-zero",);
+        ensure!(self.amount() > 0, "amount must be greater than zero",);
+        if self.bridge_address.is_some() {
+            let parsed_bridge_memo: Ics20WithdrawalFromRollupMemo =
```

Comment:
We're JSON-encoding a PB version of this type, whereas here we're decoding straight into our domain type.  The serde `try_from = "raw::Ics20WithdrawalFromRollup"` on the domain type handles that ok, but I still keep feeling uneasy when I see this.

Reply:
I could either remove the domain type or modify the serialization on the other end, it uses the pb::name property when serializing for errors which is why it uses the raw type

Reply:
It's certainly not a blocker, and maybe I'm being overly pedantic.  But I guess I'd prefer if the domain type didn't derive `serde` traits, forcing us to always use the pb type for encoding to/decoding from JSON, and we use the domain type in the business logic.

No probs if you prefer to leave as is - I think we're doing similar things elsewhere.  We can always try and get more strict on the separation of pb and domain types in the future.

Reply:
I went ahead and went back to how the memos were defined in domain type as reexporting, less code changes here a bit cleaner I think.

Reply:
I have created an issue here for follow up